src/curvefit/solvers/solvers.py

```diff
@@ -118,8 +138,11 @@ def __init__(self, gm_model, solver=None):
 
     def fit(self, data, x_init=None, options=None):
         if self.assert_solver_defined() is True:
-            self.solver.fit(data, x_init, options)
             model = self.get_model_instance()
+            if model.curve_fun.__name__ != 'gaussian_pdf':
+                raise RuntimeError('Only works with observation in gaussian pdf space.')
+            
```

Comment:
I added a check here so that only `gaussian_pdf` space is allowed. There is no data translation happening here. I think that is also the case with our old code, but correct me if that's not true @mbannick !